Mage.Sets/src/mage/cards/d/Deification.java

```diff
@@ -0,0 +1,148 @@
+package mage.cards.d;
+
+import java.util.UUID;
+
+import mage.abilities.Ability;
+import mage.abilities.common.AsEntersBattlefieldAbility;
+import mage.abilities.common.SimpleStaticAbility;
+import mage.abilities.effects.ReplacementEffectImpl;
+import mage.abilities.effects.common.ChoosePlaneswalkerTypeEffect;
+import mage.abilities.effects.common.continuous.GainAbilityAllEffect;
+import mage.abilities.keyword.HexproofAbility;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.constants.*;
+import mage.counters.CounterType;
+import mage.filter.StaticFilters;
+import mage.filter.common.FilterPlaneswalkerPermanent;
+import mage.game.Game;
+import mage.game.events.GameEvent;
+import mage.game.events.RemoveCountersEvent;
+import mage.game.permanent.Permanent;
+import org.apache.log4j.Logger;
+
+/**
+ *
+ * @author jimga150
+ */
+public final class Deification extends CardImpl {
+
+    public Deification(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.ENCHANTMENT}, "{1}{W}");
+        
+
+        // As Deification enters the battlefield, choose a planeswalker type.
+        this.addAbility(new AsEntersBattlefieldAbility(new ChoosePlaneswalkerTypeEffect(Outcome.AddAbility)));
+
+        // Planeswalkers you control of the chosen type have hexproof.
+        this.addAbility(new SimpleStaticAbility(Zone.BATTLEFIELD, new GainAbilityAllEffect(HexproofAbility.getInstance(), Duration.WhileOnBattlefield, new FilterDeification())));
+        
+        // As long as you control a creature, if damage dealt to a planeswalker you control of the chosen type would result in all loyalty counters on it being removed, instead all but one of those counters are removed.
+        addAbility(new SimpleStaticAbility(new DeificationReplacementEffect()));
+    }
+
+    private Deification(final Deification card) {
+        super(card);
+    }
+
+    @Override
+    public Deification copy() {
+        return new Deification(this);
+    }
+}
+
+// Based on Cover of Darkness
+class FilterDeification extends FilterPlaneswalkerPermanent {
```

Comment:
Why do you need a new filter class? Does a custom predicate not work?

Reply:
Oh good point lol